catch the exceptions in the current code
